code/lib/preview-api/src/modules/preview-web/PreviewWeb.test.ts

```diff
@@ -2141,39 +2141,6 @@ describe('PreviewWeb', () => {
           window.location = { ...originalLocation, reload: originalLocation.reload };
         });
 
-        it('stops initial story after loaders if running', async () => {
-          const [gate, openGate] = createGate();
-          componentOneExports.default.loaders[0].mockImplementationOnce(async () => gate);
-
-          document.location.search = '?id=component-one--a';
-          await new PreviewWeb(importFn, getProjectAnnotations).ready();
-          await waitForRenderPhase('loading');
-
-          emitter.emit(SET_CURRENT_STORY, {
-            storyId: 'component-one--b',
-            viewMode: 'story',
-          });
-          await waitForSetCurrentStory();
-          await waitForRender();
-
-          // Now let the loader resolve
-          openGate({ l: 8 });
-          await waitForRender();
-
-          // Story gets rendered with updated args
-          expect(projectAnnotations.renderToCanvas).toHaveBeenCalledTimes(1);
-          expect(projectAnnotations.renderToCanvas).toHaveBeenCalledWith(
-            expect.objectContaining({
-              forceRemount: true,
-              storyContext: expect.objectContaining({
-                id: 'component-one--b',
-                loaded: { l: 7 },
-              }),
-            }),
-            'story-element'
-          );
-        });
-
```

Comment:
@tmeasday what's your thoughts on removing this test? It's not true anymore, and superseded by the new tests.
It _could_ make sense to test that this still happens if loaders manage to finish before the magic 3 retries - but I haven't managed to get into that state here by wrangling the async tick counts exactly.

Reply:
Is it possible (now or in the future) to write a loader that is successfully aborted by the abort controller? If so I think it might be useful to have a test that checks that such a scenario doesn't break things.

Reply:
In theory, maybe. This scenario _could_ exist:

1. Remount abort rendering
1. Loader is done after 1-2 async ticks - maybe the loader does some async work with `setTimeout(..., 0)`?
2. Tearing down is done, no need to reload.

However in practice I don't know how to reproduce that - I can write a loader that finishes after 1ms but I don't have super-human reaction time to force remount a story after 1ms of loading 😅

It could happen for real though, but it would be extremely rare. A loader taking 3000ms to finish and a user triggering remount after 2999ms.

Reply:
@JReinhold I was thinking more along the lines of a loader that hooks into our abort controller (which we abort at the start of `teardown`). For instance if it calls `fetch` and passes in the abort controller. Does that apply?